templates/detail.jinja2

```diff
@@ -1,16 +1,83 @@
 {% extends "base.jinja2" %}
 {% block body %}
 
+
+
+  <aside>
+    <form action="" method="POST" class="add_entry" id="edit">
+      <div class="field">
+        <label for="title">Title</label>
+        <input type="text" size="30" name="title" id="title" value="{{ entry.title }}"/>
+      </div>
+      <div class="field">
+        <label for="text">Text</label>
+        <textarea name="text" id="text" rows="5" cols="80">{{ entry.text }}</textarea>
+      </div>
+      <div class="control_row">
+        <input type="submit" value="Share" name="Share"/>
+      </div>
+    </form>
+  </aside>
+
+
+
   <div id='wrapper'>
     <ul id="gallery">
       <article class="entry" id="{{entry.id}}">
         <h3>{{ entry.title }}</h3>
-        <p class="dateline">{{ entry.created.strftime('%b. %d, %Y') }}
-        <p class="text">{{ entry.text | safe }} </p>
+        <p class="dateline">{{ entry.created.strftime('%b. %d, %Y') }} </p>
+        <div class="text">{{ entry.display_text | safe }} </div>
         {% if request.authenticated_userid %}
-          <a href="{{ request.route_url('edit', id=entry.id) }}">Edit</a>
+        <a href="{{ request.route_url('edit', id=entry.id) }}">Edit</a>
+        <a href="https://twitter.com/share" class="twitter-share-button" data-text="" data-via="MarkSaiget">Tweet</a>
         {% endif %}
       </article>
     </ul>
   </div>
+
+  <script>
+    var twit = $(".twitter-share-button").clone();
+
+    $("#edit").hide();
+
+    $(".twitter-share-button").attr('data-text', $('#title').val())
+    $.getScript("http://platform.twitter.com/widgets.js");
+
+    $("article").children().last().prev().click( function(click) {
+      $("#edit").show();
+      $("article").hide();
+      click.preventDefault();
```

Comment:
Generally, it's best to put the `preventDefault` call *first* in the event handler so that nothing stops it from stopping the default action from taking place.